src/main/java/com/hca/car_dealership/repositories/VehicleRepository.java

```diff
@@ -0,0 +1,276 @@
+package com.hca.car_dealership.repositories;
+
+import com.hca.car_dealership.models.Vehicle;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class VehicleRepository {
+    @Autowired
+    private DataSource dataSource;
+
+    //get all vehicles
+    public List<Vehicle> getAllVehicles(int dealershipId) {
+        String query = "SELECT * FROM vehicles " +
+                "JOIN inventory ON vehicles.vehicle_id = inventory.vehicle_id " +
+                "WHERE inventory.dealership_id = ?";
+
+        List<Vehicle> vehicles = new ArrayList<>();
+        try (Connection conn = dataSource.getConnection();
+             PreparedStatement ps = conn.prepareStatement(query)) {
+            ps.setInt(1, dealershipId);
+
+            try (ResultSet rs = ps.executeQuery()) {
+                while (rs.next()) {
+                    Vehicle vehicle = mapRowToVehicle(rs);
+                    vehicles.add(vehicle);
+                }
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return vehicles;
+    }
+
+
+    public List<Vehicle> getVehicleByPriceRange(double priceLow, double priceHigh, int dealership_id) {
+        String query = "SELECT * FROM vehicles " +
+                "JOIN inventory ON vehicles.vehicle_id = inventory.vehicle_id " +
+                "WHERE vehicles.price >= ? AND vehicles.price <= ? AND inventory.dealership_id = ?";
+        List<Vehicle> vehicles = new ArrayList<>();
+        try (Connection conn = dataSource.getConnection();
+             PreparedStatement ps = conn.prepareStatement(query)) {
+            // replace question mark with value
+            ps.setDouble(1, priceLow);
+            ps.setDouble(2, priceHigh);
+            ps.setInt(3,dealership_id);
+            try (ResultSet rs = ps.executeQuery()) {
+                while (rs.next()) {
+                    Vehicle vehicle = mapRowToVehicle(rs);
+                    vehicles.add(vehicle);
+                }
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Technically we'd log this error, we wouldn't do a run time exception because I'd want the program to keep going

Reply:
I just did print stack trace in the classroom examples to simulate this